src/mrsimulator/__init__.py

```diff
@@ -42,98 +42,340 @@
+    signal_processors: List[SignalProcessor] = None
+    version: str = __version__
+    application: Dict = None
+
+    class Config:
+        validate_assignment = True
+        extra = "forbid"
+
+    @classmethod
+    def parse_dict_with_units(cls, py_dict: dict):
+        """Parse the physical quantity from a dictionary reperesentation of the
+        Mrsimulator object, where the physical quantity is expressed as a string with a
+        number and a unit
+
+        Args:
+            Dict py_dict: A required python dictionary object.
+
+        Returns:
+            A :ref:`mrsimulator_api` object.
+
+        Example
+        -------
+        >>> mrsim_dict = {
+        ...   "simulator": {
+        ...     "spin_systems": [
+        ...       {
+        ...         "sites": [
+        ...           {
+        ...             "isotope": "1H",
+        ...             "isotropic_chemical_shift": "0.0 ppm",
+        ...             "shielding_symmetric": { "zeta": "-100.0 ppm", "eta": 0.3 }
+        ...           },
+        ...           {
+        ...             "isotope": "1H",
+        ...             "isotropic_chemical_shift": "0.0 ppm",
+        ...             "shielding_symmetric": { "zeta": "-100.0 ppm", "eta": 0.3 }
+        ...           }
+        ...         ],
+        ...         "abundance": "45.0 %"
+        ...       }
+        ...     ],
+        ...     "methods": [
+        ...       {
+        ...         "name": "BlochDecaySpectrum",
+        ...         "description": "A one-dimensional Bloch decay spectrum method.",
+        ...         "channels": ["1H"],
+        ...         "spectral_dimensions": [
+        ...           {
+        ...             "count": 1024,
+        ...             "spectral_width": "25000.0 Hz",
+        ...             "events": [{ "transition_query": [{ "ch1": { "P": [-1] } }] }]
+        ...           }
+        ...         ],
+        ...         "magnetic_flux_density": "9.4 T",
+        ...         "rotor_angle": "0.9553166181245 rad",
+        ...         "rotor_frequency": "0.0 Hz"
+        ...       }
+        ...     ],
+        ...     "config": {
+        ...       "number_of_sidebands": 64,
+        ...       "integration_volume": "octant",
+        ...       "integration_density": 70,
+        ...       "decompose_spectrum": "none"
+        ...     }
+        ...   },
+        ...   "signal_processors": [
+        ...     {
+        ...       "operations": [
+        ...         { "dim_index": 0, "function": "IFFT" },
+        ...         {
+        ...           "dim_index": 0,
+        ...           "FWHM": "2500.0 Hz",
+        ...           "function": "apodization",
+        ...           "type": "Exponential"
+        ...         },
+        ...         { "dim_index": 0, "function": "FFT" },
+        ...         { "factor": 20.0, "function": "Scale" }
+        ...       ]
+        ...     }
+        ...   ],
+        ...   "version": "0.7.0dev1",
```

Comment:
You are comparing version number here.

Reply:
Passing version as a string has been removed in examples and tests